### DISCUSSION TITLE
Caddy Rate Limit Customization

### DISCUSSION THREAD

**Discussion by JaronrH** [Kovah/LinkAce#857] [Ideas & Feature Requests]
I'm using v2.0 beta that runs on Caddy within a docker image.  I'm running into an issue where the API is getting rate-limited VERY quickly whenever I do batch operations against it!   **_Can you add the ability to change the default Rate Limiting?_**  Thanks!

**Comment by Kovah** [comment]
Hi, thanks for taking the time to report this issue. Could you share more details about the volume of requests you are sending, as well as an example response with the full headers?

**Comment by JaronrH** [comment]
Sure!  Here is an example:
I've hit /api/v1/lists?per_page=-1 60 times in less than 1 minute.  This has triggered the rate limit protection and return a Status 429.

Headers:
![image](https://github.com/user-attachments/assets/55839b2d-01c7-4c42-8429-140ca6ffba11)

Response:
```
{
    "message": "Too Many Attempts."
}
```

**Comment by Kovah** [comment]
What do you think would be a good limit for API calls, i.e. how many calls were you trying to make?

**Comment by JaronrH** [comment]
I don't know if there is a one-size-fits-all answer!  Some people are running LinkAce on a PI while others may be on using actual servers which means that both would handle the request volume differently.  For this reason, I would think using environment variables would make more sense.

Ex:  
Variables:
```
export RATE_LIMIT="10r/s"
export BURST_LIMIT="20"
```

Caddyfile:
```
...
:{$PORT:80} {
    limit_req rate=${RATE_LIMIT} burst=${BURST_LIMIT} nodelay
    root * /app/public
    ....
}
```

(In my specific case, a bulk API that lets me download links from multiple lists at once would be helpful but that's another discussion entirely!)

**Comment by Kovah** [comment]
Sure there's no answer for everybody, but a more sensible default would be helpful.

Also, this has nothing to do with Caddy. It's in the core application. There will be a API_RATE_LIMIT variable for this with the final release.

**Comment by JaronrH** [comment]
Honestly, I think the current 60/m rate is a great default API limit.  Worst case, jumping up to 100/m as the default wouldn't be bad either but I don't see a compelling reason to go any higher than that as the default.
